src/lib/CookieJar.js

```diff
@@ -38,7 +38,9 @@ class CookieJar { // eslint-disable-line no-unused-vars
      * @returns {string}
      */
     static _encodeCookie(keys) {
-        return keys.map(keyInfo => `${keyInfo.type}${keyInfo.encrypted ? 1 : 0}${keyInfo.id}`).join('');
+        return keys.map(
+            keyInfo => `${keyInfo.type}${keyInfo.encrypted ? 1 : 0}${keyInfo.hasPin ? 1 : 0}${keyInfo.id}`,
```

Comment:
As discussed on the phone, we will rather try to improve cookie storage capacity in the Accounts Manager, since that needs to transfer much more data in the cookie, thus will be the bottle neck.

Improving this for the sake of readability would be premature optimization.